### ISSUE TITLE
Valid usecase for `reduce()`

### ISSUE THREAD

**Issue by tylerlaprade** [biomejs/biome#1814]
### Environment information

```bash
N/A
```

### Rule name

`noParameterAssign`, `noAssignInExpressions`

### Playground link

https://biomejs.dev/playground/?code=ZgBvAG8ALgByAGUAZAB1AGMAZQAoACgAcAByAGUAdgBpAG8AdQBzAFYAYQBsAHUAZQAsACAAZQB4AGkAcwB0AGkAbgBnAFYAYQBsAHUAZQApACAAPQA%2BACAAKABwAHIAZQB2AGkAbwB1AHMAVgBhAGwAdQBlACAAKwA9ACAAZQB4AGkAcwB0AGkAbgBnAFYAYQBsAHUAZQApACwAIAAwACkAOwA%3D

### Expected result

This is the only way `.reduce()` can realistically be used. I don't want to disable these rules for other usecases in my project, and I don't want to manually disable every line that uses `.reduce()`.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
I don't understand why you need to re-assign the parameter here.
You could write `foo.reduce((previousValue, existingValue) => (previousValue + existingValue), 0);` right?

**Comment by tylerlaprade** [comment]
Ooh, I think you're right! I ran across this code in my codebase and didn't check how reduce actually works, my apologies. Thank you!
